code/game/objects/items/weapons/twohanded.dm

```diff
@@ -421,6 +421,78 @@
 			M.Copy_Parent(user, 100, user.health/2.5, 12, 30)
 			M.GiveTarget(L)
 
+/obj/item/weapon/twohanded/ram
+	name = "kinetic battering ram"
+	desc = "Internal gyroscopes give this ram excellent balance and the necessary force to break doors... or ribs."
+	icon_state = "ram0"
+	force = 10
+	force_wielded = 24
+	w_class = 4
+	throwforce = 15
+	throw_speed = 2
+	throw_range = 3
+	slot_flags = SLOT_BACK
+	materials = list(MAT_METAL=13000)
+	origin_tech = "materials=2;engineering=2;combat=2"
+	attack_verb = list("crushed", "bashed", "bludgeoned", "smashed", "pulverized")
+	hitsound = "swing_hit"
+	var/removingairlock = 0
+
+/obj/item/weapon/twohanded/ram/update_icon()
+	icon_state = "ram[wielded]"
+	return
+
+/obj/item/weapon/twohanded/ram/suicide_act(mob/user)
+	user.visible_message("<span class='suicide'>[user] swings [src] into their face! It looks like they're trying to commit suicide..</span>")
+	playsound(src,'sound/effects/bang.ogg',50,1)
+	return (BRUTELOSS)
+
+/obj/item/weapon/twohanded/ram/afterattack(atom/T, mob/user, proximity)
+	if(!proximity)
+		return
+	if(!wielded)
+		src << "<span class='warning'>You need both hands to break down a door!</span>"
+	if(istype(T, /obj/machinery/door/airlock))
+		if(!removingairlock)
+			var/obj/machinery/door/airlock/A = T
+			removingairlock = 1
+			src << "<span class='notice'>You begin to break the door down...</span>"
+			playsound(A.loc, 'sound/effects/bang.ogg', 75, 1)
+			user.do_attack_animation(A)
+			spawn(20)
+				for(var/i in 1 to 3)
+					if(i == 1)
+						if(!proximity)
+							return
+						playsound(A.loc, 'sound/effects/bang.ogg', 50, 1)
+						user.do_attack_animation(A)
+						sleep(20)
+					if(i == 2)
+						if(!proximity)
+							return
+						playsound(A.loc, 'sound/effects/bang.ogg', 75, 1)
+						playsound(A.loc, 'sound/machines/airlock_alien_prying.ogg', 100, 1)
+						var/datum/effect_system/spark_spread/sparks = new
+						sparks.set_up(3, 1, A.loc) //no idea what the 0 is
```

Comment:
do not try to bend the zero, that's impossible. instead, only try to realize the truth...